### DISCUSSION TITLE
Can a primitive atom hold a function?

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1238] [Q&A] [answered]
I'd like to store a function in a primitive atom, but Jotai wants to call this function.  Can functions be stored in an atom?

What is the intended use of functions passed to atom(fn)?

This works: atom(() => fn)

**Comment by dai-shi** [comment, marked as answer]
You need to wrap with an object.
```js
const fnAtom = atom({ fun: () => 'hello' })
```



> This works: atom(() => fn)

It doesn't create a primitive atom. It creates a read-only atom.
